test/apps/iodemo/io_demo.cc

```diff
@@ -2419,6 +2429,34 @@ class DemoClient : public P2pDemoCommon {
     MemoryPool<IoReadResponseCallback>      _read_callback_pool;
 };
 
+int set_sockaddr(const char *ip_str, uint16_t port, struct sockaddr *addr)
```

Comment:
1. make this function part of DemoClient class
2. pass `const std::string&` as first parameter instead of `const char*`
3. rename 'addr' to 'saddr'

Reply:
changed accordingly